strangerstudios/paid-memberships-pro
# Add cron helper functions and implement filter to hook into to register Stripe sub cron

### All Submissions:






### Changes proposed in this Pull Request:



Resolves issues with missing crons by scheduling them during cron-ready checks.

### How to test the changes in this Pull Request:

1. Load up the plugin
2. Use a cron helper plugin that shows cron runs and schedule
3. Delete your `cron` option in the DB
4. Verify PMPro crons are not showing up on page refresh
5. On next cron ready check, the crons should be scheduled again as expected

### Other information:




### Changelog entry

```
* ENHANCEMENT: Introduced a new set of functions that handle cron-related tasks including: `pmpro_get_crons()` to get the list of PMPro registered crons and `pmpro_maybe_schedule_cron()` which supersedes the `pmpro_maybe_schedule_event()` function now. #1999 (@sc0ttkclark)
* ENHANCEMENT: New filter `pmpro_registered_crons` which you can register new crons to be handled by PMPro. They show up in the PMPro Site Health info and are automatically scheduled when they need to be. #1999 (@sc0ttkclark)
* BUG FIX: Crons are now automatically rescheduled if they disappear form the cron schedule. #1999 (@sc0ttkclark, @mircobabini)
```
